reconcile/utils/oc.py

```diff
@@ -529,9 +540,11 @@ def sa_get_token(self, namespace, name):
     def get_api_resources(self):
         # oc api-resources only has name or wide output
         # and we need to get the KIND, which is the last column
-        cmd = ["api-resources", "--no-headers"]
-        results = self._run(cmd).decode("utf-8").split("\n")
-        return [r.split()[-1] for r in results]
+        if not self.api_resources:
```

Comment:
this is not thread safe and i would like you to have a closer look if that could be an issue. i know that we do a lot of threaded executions when it comes to the openshift-* integrations. as long as we don't process resources from the same cluster in multiple threads, we should be fine

Reply:
This did indeed cause issues: https://github.com/app-sre/qontract-reconcile/pull/2870